evm_loader/program/src/executor/state.rs

```diff
@@ -449,20 +453,39 @@ impl<'a, B: AccountStorage> Database for ExecutorState<'a, B> {
             .collect::<Vec<_>>();
 
         if !metas.iter().any(|m| (m.pubkey == address) && m.is_writable) {
+            // account with given address don't changed by current actions
+            // put offset to cache
+            self.data.cache.borrow_mut().actions_offset = self.data.actions.len();
```

Comment:
Other accounts may be affected by current actions.
`.actions_offset` should only be changed after actions have been emulated.